test/join.jl

```diff
@@ -1557,4 +1559,163 @@ end
                     c="c", d="d")
 end
 
+@testset "matchmissing :notequal correctness" begin
+    Random.seed!(1337)
+    names = [
+        DataFrame(ID = Union{Int, Missing}[1, 2, missing],
```

Comment:
`Union{Int, Missing}` isn't needed here. Also, below, what's the point of using `Union{T, Missing}` even in the absence of `missing` values?

Reply:
we check for `Missing` in the eltype in `completecases` so it matters (but it is not super crucial as we have `completecases` test coverage done properly).